store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/impl/FederatedOperationHandler.java

```diff
@@ -140,4 +161,31 @@ private List<GraphSerialisable> getGraphs(final FederatedOperation<INPUT, OUTPUT
                 graphs
                 : Collections.emptyList();
     }
+
+    /**
+     * Checks if the schemas of the supplied graphs have any of the same groups.
+     *
+     * @param graphs The list of graphs.
+     * @return True if any schema groups are shared.
+     */
+    private boolean graphsHaveCommonSchemaGroups(final List<GraphSerialisable> graphs) {
+        // Check if any of the graphs have common groups in their schemas
+        List<Schema> graphSchemas = graphs.stream()
+            .map(GraphSerialisable::getSchema)
+            .collect(Collectors.toList());
+
+        // Compare all schemas against each other
+        for (int i = 0; i < graphSchemas.size() - 1; i++) {
+            for (int j = i + 1; j < graphSchemas.size(); j++) {
+                // Compare each schema against the others to see if common groups
+                Set<String> firstGroupSet = graphSchemas.get(i).getGroups();
+                Set<String> secondGroupSet = graphSchemas.get(j).getGroups();
+                if (firstGroupSet.stream().anyMatch(secondGroupSet::contains)) {
```

Comment:
There's no reason to use C style looping here.
```suggestion
        for (Schema schemaLeft : graphSchemas) {
            for (Schema schemaRight : graphSchemas) {
                // Compare each schema against the others to see if common groups
                Set<String> groupSetLeft = schemaLeft.getGroups();
                Set<String> groupSetRight = schemaRight.getGroups();
                if (groupSetLeft.stream().anyMatch(groupSetRight::contains)) {
```

Reply:
This will do duplicate comparisons e.g. if you have `[1,2,3]` and loop like this you will end up comparing both directions:
```
[1,1]
[1,2]
[1,3]
```
and
```
[1,1]
[2,1]
[3,1]
``

Reply:
Ah yes, had not looked at the loop properly. I think there's no easy way to do this in Java 8, which is a shame. Could use `Collections.disjoint` instead of `stream`?

Reply:
Yeah it is unfotunate, good point `disjoint` might be bit cleaner to use here.